Sklep/MainWindow.cs

```diff
@@ -21,6 +35,8 @@ private void zarejestrujNowyProduktToolStripMenuItem_Click(object sender, EventA
 
         private void MainWindow_Load(object sender, EventArgs e)
         {
+            db = new DatabaseContext();
```

Comment:
Aktualnie nigdzie nie zamykasz tego kontekstu. Przedłużanie jego żywotności bez powodu też nie jest dobrą praktyką. Domyślam się, że nie chciałeś z każdym nowym produktem wczytywać listy produktów. Spójrz na ten kod:
```
using (var context = new SchoolDBEntities())
{
    var query = context.Students
                       .where(s => s.StudentName == "Bill")
                       .FirstOrDefault<Student>();
}
```
Tutaj dużo ładniej zarządzamy obiektem kontekstu bazy danych, select nadal jest w LINQ (to tam niżej, linia 61). No i nie musimy wczytywać wszystkich produktów, żeby pobrać 1